crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -436,6 +438,31 @@ fn path() -> impl NoirParser<Path> {
     ))
 }
 
+fn path() -> impl NoirParser<Path> {
+    path0(false)
+}
+
+fn rename() -> impl NoirParser<Option<Ident>> {
+    ignore_then_commit(keyword(Keyword::As), ident()).or_not()
+}
+
+fn path_simple() -> impl NoirParser<UseTree> {
+    path().then(rename()).map(|(name, rename)| UseTree::Simple(name, rename))
+}
+
+fn path_list() -> impl NoirParser<UseTree> {
+    let path = path0(true);
+    let items = path_list_item()
+        .separated_by(just(Token::Comma))
+        .delimited_by(just(Token::LeftBrace), just(Token::RightBrace));
+
+    path.then(items).map(|(path, names)| UseTree::List(path, names))
```

Comment:
The `allow_trailing` doesn't work here since it'd allow for `use foo bar::baz;` being parsed as `use foo::bar::baz;`. I think we'll need to separate the old path parser from a new recursive parser to parse UseTrees